WeatherService.Test/WeatherReportPublishingTests/WeatherReportPublisherTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using AutoFixture.Xunit2;
+using FluentAssertions;
+using Moq;
+using WeatherMonitoringAndReportingService.Models;
+using WeatherMonitoringAndReportingService.WeatherReportPublishing;
+
+namespace WeatherService.Test.WeatherReportPublishingTests
+{
+    public class WeatherReportPublisherTests
+    {
+        [Theory]
+        [AutoData]
+        public void Subscribe_AddsObserverToWeatherObserversList(
+            Mock<IWeatherObserver> mockObserver,
+            WeatherReportPublisher sut)
+        {
+            //Act
+            sut.Subscribe(mockObserver.Object);
+
+            //Assert
+            sut.GetSubscribers().Should().HaveCount(1);
+            sut.GetSubscribers().Should().Contain(mockObserver.Object);
+        }
+
+        [Theory]
+        [AutoMoqData]
+        public void Subscribe_AddsObserversToWeatherObserversList(
+            List<IWeatherObserver> mockObservers,
+            WeatherReportPublisher sut)
+        {
+            //Act
+            sut.Subscribe(mockObservers);
+
+            //Assert
+            sut.GetSubscribers().Should().Contain(mockObservers);
+            sut.GetSubscribers().Should().HaveSameCount(mockObservers);
+        }
+
+        [Theory]
+        [AutoMoqData]
+        public void Unsubscribe_RemoveObserverFromWeatherObserversList(
+            Mock<IWeatherObserver> mockObserver,
+            WeatherReportPublisher sut)
+        {
+            //Arrange
+            sut.Subscribe(mockObserver.Object);
+
+            //Act
+            sut.Unsubscribe(mockObserver.Object);
+
+            //Assert
+            sut.GetSubscribers().Should().NotContain(mockObserver.Object);
```

Comment:
let's discuss how we test this in a different way without depending on GetSubscribers